ipaserver/install/plugins/ca_renewal_master.py

```diff
@@ -109,16 +109,18 @@ def execute(self, **options):
             logger.debug("certmonger request for RA cert not found")
 
             config = directivesetter.get_directive(
-                paths.CA_CS_CFG_PATH, 'subsystem.select', '=')
+                paths.CA_CS_CFG_PATH, 'ca.crl.MasterCRL.enableCRLUpdates', '=')
 
-            if config == 'New':
+            if config is None or config.lower() == 'true':
+                # CA is a renewal master
                 pass
-            elif config == 'Clone':
+            elif config.lower() == 'false':
+                # CA is not a renewal master
                 return False, []
             else:
                 logger.warning(
-                    "CS.cfg has unknown subsystem.select value '%s', "
-                    "assuming local CA is not a renewal master", config)
+                    "CS.cfg has unknown ca.crl.MasterCRL.enableCRLUpdates value"
+                    " '%s', assuming local CA is not a renewal master", config)
```

Comment:
This code is trying to configure the initial IPA CA to be the renewal master if one is not already set. If there is no role with caRenewalMaster set then it relies on subsystem.select to be 'New' which again, is unreliable (but I suppose better than nothing). If we get into this else then it means that either there is no role as renewal master, that the IPA RA is not tracked or it is tracked but has no ca-name!? 

So in this case, I think that if we get this far and have not identified a machine as the renewal master then defaulting to the machine that is the CRL generator is about as good as it's going to get.

I think this code will work at least as well as the code it's replacing. It would be great to get a second opinion.

I think this would only apply to servers upgraded from pre 4.0. It is doubtful this will be all that impactful as any upgrade between 4.0 and 4.11 would already have this corrected.

Reply:
> This code is trying to configure the initial IPA CA to be the renewal master if one is not already set. If there is no role with caRenewalMaster set then it relies on subsystem.select to be 'New' which again, is unreliable (but I suppose better than nothing). If we get into this else then it means that either there is no role as renewal master, that the IPA RA is not tracked or it is tracked but has no ca-name!?
> 
Agree, we get into this code only if multiple things were checked before and are inconclusive.

> So in this case, I think that if we get this far and have not identified a machine as the renewal master then defaulting to the machine that is the CRL generator is about as good as it's going to get.
> 
> I think this code will work at least as well as the code it's replacing. It would be great to get a second opinion.
> 

Agree. But instead of reading the CS.cfg file the code could use the method ca.is_crlgen_enabled() that returns True/False or raises an Exception when the config is inconsistent.

> I think this would only apply to servers upgraded from pre 4.0. It is doubtful this will be all that impactful as any upgrade between 4.0 and 4.11 would already have this corrected.

Reply:
@edewata 
could you replace this part with a call to ca.is_crlgen_enabled()? Is it functionally equivalent but reuses common code.

Reply:
Sure, please see the updated patch.